File: PetShop.Model/Recenzija.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace PetShop.Model
{
    public class Recenzija
    {
        public int Id { get; set; }
        public int Ocjena { get; set; }
        public DateTime Datum { get; set; }
        public int KorisnikId { get; set; }
        public int ProizvodId { get; set; }

        public virtual Korisnik Korisnik { get; set; }
        public virtual Proizvod Proizvod { get; set; }
    }
}

File: petshop_mobile/lib/models/Recenzija.dart
class Recenzija {
  final int? id;
  final int? ocjena;
  final DateTime? datum;
  final int? korisnikId;
  final int? proizvodId;

  Recenzija({
    required this.id,
    required this.ocjena,
    required this.datum,
    required this.korisnikId,
    required this.proizvodId,
  });

  factory Recenzija.fromJson(Map<String, dynamic> json) {
    return Recenzija(
      id: int.parse(json["id"].toString()),
      ocjena: json["ocjena"],
      datum: DateTime.tryParse(json['datum']),
      korisnikId: int.parse(json["korisnikId"].toString()),
      proizvodId: int.parse(json["proizvodId"].toString()),
    );
  }

  Map<String, dynamic> toJson() => {
        "id": id,
        "ocjena": ocjena,
        "datum":
            datum == null ? null : datum!.toIso8601String(),
        "korisnikId": korisnikId,
        "proizvodId": proizvodId,
      };
}
